instrumentation/nats/nats-2.17/library/src/main/java/io/opentelemetry/instrumentation/nats/v2_17/internal/NatsMessageWritableHeaders.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.nats.v2_17.internal;
+
+import io.nats.client.Message;
+import io.nats.client.impl.Headers;
+import io.nats.client.impl.NatsMessage;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class NatsMessageWritableHeaders {
+
+  public static Message create(String subject, byte[] body) {
+    System.out.println(new Headers());
+    System.out.println(new Headers().isReadOnly());
+    return NatsMessage.builder().subject(subject).headers(new Headers()).data(body).build();
+  }
+
+  public static Message create(String subject, String replyTo, byte[] body) {
+    return NatsMessage.builder()
+        .subject(subject)
+        .replyTo(replyTo)
+        .headers(new Headers())
+        .data(body)
+        .build();
+  }
+
+  public static Message create(String subject, Headers headers, byte[] body) {
+    if (headers == null || headers.isReadOnly()) {
+      headers = new Headers(headers);
+    }
+    return NatsMessage.builder().subject(subject).headers(headers).data(body).build();
+  }
+
+  public static Message create(String subject, String replyTo, Headers headers, byte[] body) {
+    if (headers == null || headers.isReadOnly()) {
+      headers = new Headers(headers);
+    }
+    return NatsMessage.builder()
+        .subject(subject)
+        .replyTo(replyTo)
+        .headers(headers)
+        .data(body)
+        .build();
+  }
+
+  public static Message create(Message message) {
+    if (message instanceof NatsMessage
```

Comment:
Not 100% sure about this. I didn't want to override all possible implementations of Message. My guess is that if there is a custom implementation not being the default, there might be something custom at the Connection level and we might better do nothing on this one